src/inmanta/module.py

```diff
@@ -1120,8 +1168,21 @@ class ProjectMetadata(Metadata, MetadataFieldRequires):
       project.yml.
     :param freeze_operator: (Optional) This key determines the comparison operator used by the freeze command.
       Valid values are [==, ~=, >=]. *Default is '~='*
+    :param type_hints: [EXPERIMENTAL FEATURE] A list of type hints that indicate the order in which the compiler
+                       should freeze list. A type hint should be specified using the following format:
+                       `<first-type>.<relation-name> before <then-type>.<relation-name>`. With this rule in place,
+                       the compiler will first freeze `first-type.relation-name` and only then `then-type.relation-name`.
     """
 
+    _re_identifier = r"[a-zA-Z_][a-zA-Z_0-9-]*"
```

Comment:
Please pay specific attention to this during the review. I am not entirely sure whether this is correct.

Reply:
it should match what is in the lexer, I think we don't allow `-` anymore